### PR TITLE
[Enhancement] Support jdk17 in docker image

### PR DESCRIPTION
**Pull request by gengjun-git**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by gengjun-git** [comment]
> This means we will use jdk17 as the default jdk for v3.5 onward?

yes

**Comment by kevincai** [comment]
@mergify rebase

File: docker/dockerfiles/toolchains/toolchains-centos7.Dockerfile [hunk 1]
```diff
@@ -70,7 +70,7 @@ LABEL org.opencontainers.image.source="https://github.com/starrocks/starrocks"
 
 RUN yum-config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo && yum install -y gh && \
         yum install -y ${EPEL_RPM_URL} && yum install -y wget unzip bzip2 patch bison byacc flex autoconf automake make \
-        libtool which git ccache binutils-devel python3 file java-11-openjdk java-11-openjdk-devel java-11-openjdk-jmods less psmisc && \
+        libtool which git ccache binutils-devel python3 file java-17-openjdk java-17-openjdk-devel java-17-openjdk-jmods less psmisc && \
```
Comment (kevincai) [inline docker/dockerfiles/toolchains/toolchains-centos7.Dockerfile:73]: jdk17 is not there in centos7 yum repository. It is not good enough just bumping java-11 to java-17

Comment (gengjun-git) [reply docker/dockerfiles/toolchains/toolchains-centos7.Dockerfile:73]: So what's your advice? Download from source?

Comment (kevincai) [reply docker/dockerfiles/toolchains/toolchains-centos7.Dockerfile:73]: if can't find any yum repository to download the jdk17, it makes no option but downloads from official site.

File: docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
 
+ARG JDK_INSTALL_HOME=/opt/jdk17
+
 FROM ubuntu:22.04
 
 # Install common libraries and tools that are needed for dev environment
 RUN apt-get update -y && \
     apt-get install --no-install-recommends -y \
-    automake binutils-dev bison byacc ccache flex libiberty-dev libtool maven zip python3 python-is-python3 make cmake gcc g++ default-jdk git patch lld bzip2 \
+    automake binutils-dev bison byacc ccache flex libiberty-dev libtool maven zip python3 python-is-python3 make cmake gcc g++ git patch lld bzip2 \
```
Comment (kevincai) [inline docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile:11]: jdk17 should be available in ubuntu platform with apt-get

Comment (gengjun-git) [reply docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile]: done

**Review by kevincai — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**